src/Model/Term.php

```diff
@@ -252,64 +315,82 @@ public static function populateTagsFromItem(&$item)
 			'tags' => [],
 			'hashtags' => [],
 			'mentions' => [],
+			'implicit_mentions' => [],
 		];
 
 		$searchpath = System::baseUrl() . "/search?tag=";
 
 		$taglist = DBA::select(
 			'term',
 			['type', 'term', 'url'],
-			["`otype` = ? AND `oid` = ? AND `type` IN (?, ?)", TERM_OBJ_POST, $item['id'], TERM_HASHTAG, TERM_MENTION],
+			['otype' => self::OBJECT_TYPE_POST, 'oid' => $item['id'], 'type' => [self::HASHTAG, self::MENTION, self::IMPLICIT_MENTION]],
 			['order' => ['tid']]
 		);
-
 		while ($tag = DBA::fetch($taglist)) {
 			if ($tag['url'] == '') {
 				$tag['url'] = $searchpath . rawurlencode($tag['term']);
 			}
 
 			$orig_tag = $tag['url'];
 
-			$author = ['uid' => 0, 'id' => $item['author-id'],
-				'network' => $item['author-network'], 'url' => $item['author-link']];
-
-			$prefix = '';
-			if ($tag['type'] == TERM_HASHTAG) {
-				$tag['url'] = Contact::magicLinkByContact($author, $tag['url']);
-				if ($orig_tag != $tag['url']) {
-					$item['body'] = str_replace($orig_tag, $tag['url'], $item['body']);
-				}
+			$prefix = self::TAG_CHARACTER[$tag['type']];
```

Comment:
Are you sure that there aren't any missmatchings `$tag['type']` and `self::TAG_CHARACTER[]`? Because this seems to possibly create PHP errors/notices.

Reply:
The queried term types are narrowly defined in the `$taglist` DBA call, so I'm not worried about this.